apis/crd/v1alpha1/cloudprovideraccount_webhook.go

```diff
@@ -126,7 +139,72 @@ func (r *CloudProviderAccount) ValidateCreate() error {
 func (r *CloudProviderAccount) ValidateUpdate(old runtime.Object) error {
 	cloudprovideraccountlog.Info("validate update", "name", r.Name)
 
-	// TODO(user): fill in your validation logic upon object update.
+	cloudProviderType, err := r.GetAccountProviderType()
+	if err != nil {
+		return err
+	}
+	switch cloudProviderType {
+	case AWSCloudProvider:
+		awsConfig := r.Spec.AWSConfig
+
+		// warning for using role based auth
+		if len(strings.TrimSpace(awsConfig.RoleArn)) != 0 {
```

Comment:
Can you make a function and use it for both validate create and update? rather than copying it twice